keps/sig-network/4770-endpointslice-controller-flexibility/kep.yaml

```diff
@@ -0,0 +1,48 @@
+title: EndpointSlice Controller Flexibility
+kep-number: 4770
+authors:
+  - "@LionelJouin"
+owning-sig: sig-network
+participating-sigs:
+  - sig-network
+status: provisional
+creation-date: 2024-07-19
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+see-also:
+  - "/keps/sig-network/2447-Make-kube-proxy-service-abstraction-optional"
+  - "/keps/sig-network/3685-endpointslice-reconciler-to-staging"
+  - "https://github.com/kubernetes/enhancements/issues/3698"
+  - "https://github.com/kubernetes/enhancements/pull/1561"
+  - "https://github.com/kubernetes/kubernetes/pull/118953#discussion_r1245970845"
+  - "https://github.com/kubernetes/endpointslice?tab=readme-ov-file#purpose"
+  - "https://github.com/kubernetes/kubernetes/issues/87412"
+  - "https://github.com/kubernetes/kubernetes/pull/87488"
+  - "https://github.com/kubernetes/enhancements/pull/1561"
+replaces: []
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: "v1.32"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "v1.32"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: EndpointControllerNameWellKnownLabel
```

Comment:
I suggest another name, `ExternalEndpointSlicesController` or similar

Reply:
I renamed it to `ExternalEndpointController` (to be aligned with the label `service.kubernetes.io/endpoint-controller-name`)